scripts/apps/search/directives/SavedSearchManageSubscribers.ts

```diff
@@ -208,24 +214,26 @@ export function SavedSearchManageSubscribers(asset, userList, api, modal, desks)
                 modal.confirm(
                     gettext('Are you sure to remove this subscription?'),
                     gettext('Unsubscribe user'),
-                ).then(() => {
-                    unsubscribeUser(
+                ).then(async () => {
```

Comment:
Why do you use async/await here?

In order to use async/await, the function you are awaiting has to return a promise. If it returns a promise and you want to perform an action after it resolves, using `then` is all you need.

```
modal.confirm(
    gettext('Are you sure to remove this subscription?'),
    gettext('Unsubscribe user'),
).then(unsubscribeUser(
    scope.savedSearch,
    user._id,
    api,
)).then(() => {
    scope.onSubscriptionsChange(updatedSearch);
});
```

Also, async/await is not used in the codebase yet, and I think the code is more readable if we stick to one way of doing things. We could start using async/await as well if it has significant advantages. Does it? I haven't looked deeply into it myself.

Reply:
async/await is just syntactic sugar for promises. I can change it but there's no need to choose between `then` or `async/await`, they can live together. Actually I did have the code you're showing and then changed it to await cause it's simpler IMO but I don't care about it much, both are readable

Reply:
It is similar to syntactic sugar, but not entirely. If you check the polyfills, it's built on es6 generators. The concept itself of writing async code inline is quite different. I'd say let's stick to promises for now since that's a uniform way of doing async in the codebase. We can add async/await later if we bump into a use case that's hard to do with promises.